src/services/clipService.ts

```diff
@@ -220,3 +323,119 @@ export async function computeTextEmbedding(
         throw err;
     }
 }
+
+export async function computeONNXTextEmbedding(
+    text: string
+): Promise<Float32Array> {
+    try {
+        const imageSession = await getOnnxTextSession();
+        const tokenizer = getTokenizer();
+        const tokenizedText = Int32Array.from(tokenizer.encodeForCLIP(text));
+        const feeds = {
+            input: new ort.Tensor('int32', tokenizedText, [1, 77]),
+        };
+        const results = await imageSession.run(feeds);
+        const embedVec = results['output'].data; // Float32Array
+        return normalizeEmbedding(embedVec);
+    } catch (err) {
+        if (err.message === CustomErrors.MODEL_DOWNLOAD_PENDING) {
+            log.info(CustomErrors.MODEL_DOWNLOAD_PENDING);
+        } else {
+            logErrorSentry(err, 'Error in computeTextEmbedding');
+        }
+        throw err;
+    }
+}
+
+async function getRGBData(inputFilePath: string) {
+    const jpegData = await readFile(inputFilePath);
```

Comment:
try catch the possibility of the `inputFilePath` not existing on disk?

Reply:
I have added a check validate that inputFilePath exists and if the it doesn't throw an error. 

But we can't recover if either of `inputFilePath doensn't existing` , `readFile` fails or `jpeg.decode` fails, so haven't added any try catch around on the whole getRGBData function. if they fails they will break the flow of getRGBData, which is intented behaviour.`

Reply:
I have added the `inputFilePath` validation directly to the getImageEmbedding function